tests/test_docs/test_orm_integration/test_orm_integration.py

```diff
@@ -0,0 +1,382 @@
+      total_price: 10
+      seller_tx_fee: 0
+      currency_id: 'FET'
+      ledger_id: 'fetchai'
+      is_ledger_tx: True
+      has_data_source: True
+      data_for_sale: {}
+      search_schema:
+        attribute_one:
+          name: country
+          type: str
+          is_required: True
+        attribute_two:
+          name: city
+          type: str
+          is_required: True
+      search_data:
+        country: UK
+        city: Cambridge
+dependencies:
+  SQLAlchemy: {}"""
+
+buyer_strategy_replacement = """models:
+  strategy:
+    class_name: Strategy
+    args:
+      max_price: 40
+      max_buyer_tx_fee: 100
+      currency_id: 'FET'
+      ledger_id: 'fetchai'
+      is_ledger_tx: True
+      search_query:
+        search_term: country
+        search_value: UK
+        constraint_type: '=='
+ledgers: ['fetchai']"""
+
+
+ORM_SELLER_STRATEGY_PATH = Path(
+    ROOT_DIR, "tests", "test_docs", "test_orm_integration", "orm_seller_strategy.py"
+)
+
+
+class TestOrmIntegrationDocs:
+    """This class contains the tests for the orm-integration.md guide."""
+
+    @pytest.fixture(autouse=True)
+    def _start_oef_node(self, network_node):
+        """Start an oef node."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set the tests up."""
+        cls.tmpdir = tempfile.mkdtemp()
+        cls.oldcwd = os.getcwd()
+        os.chdir(cls.tmpdir)
+        cls.runner = CliRunner()
+
+        cls.aea_seller_name = "my_seller_aea"
+        cls.aea_buyer_name = "my_buyer_aea"
+        cls.aea_seller_dir = Path(cls.tmpdir, cls.aea_seller_name)
+        cls.aea_buyer_dir = Path(cls.tmpdir, cls.aea_buyer_name)
+
+        # copy 'packages/' into temporary test directory
+        shutil.copytree(Path(ROOT_DIR, "packages"), Path(cls.tmpdir, "packages"))
+
+        markdown_parser = mistune.create_markdown(renderer=mistune.AstRenderer())
+
+        skill_doc_file = Path(ROOT_DIR, "docs", "orm-integration.md")
+        doc = markdown_parser(skill_doc_file.read_text())
+        # get only code blocks
+        code_blocks = list(filter(lambda x: x["type"] == "block_code" == "python", doc))
+        cls.python_code_blocks = list(
+            filter(lambda x: x["info"].strip() == "python" == "python", code_blocks)
+        )
+
+    def test_strategy_consistency(self):
+        """
+        Test that the seller strategy specified in the documentation
+        is the same we use in the tests.
+        """
+        strategy_file_content = ORM_SELLER_STRATEGY_PATH.read_text()
+        for python_code_block in self.python_code_blocks:
+            if not python_code_block["text"] in strategy_file_content:
+                pytest.fail(
+                    "Code block not present in strategy file:\n{}".format(
+                        python_code_block["text"]
+                    )
+                )
+
+    def test_example(self):
+        """Test the example."""
+        # # init the CLI tool locally
+        result = self.runner.invoke(
```

Comment:
Ok I suggest using the AEA test case, otherwise we need to go back and change this later. Sorry, but the benefit is big (much fewer lines of code, easier extension)

Reply:
See for an example: https://github.com/fetchai/agents-aea/blob/develop/tests/test_packages/test_skills/test_carpark.py